projects/vdk-core/src/vdk/internal/core/errors.py

```diff
@@ -146,116 +238,92 @@ def to_html(self) -> str:
         return self._to_string(self._get_template("<br />"))
 
 
-def get_blamee_overall() -> str | None:
+def get_blamee_overall(
```

Comment:
I find the method over-complicated. Can we simplify it. Do we need a predicate ?  

I don't understand its purpose.

Reply:
the reason for introducing the predicate, is backwards-compatibility with older usages, that do not need any object pre-filtering, before returning the object property, simply overloading. I think it's the simplest way, since it does not require any rewriting of old code and usages, yet brings new functionality. The alternative is, to introduce multiple methods that call eachother -> to achieve just the same with one optional predicate argument

Reply:
What older usages? This should not be used outside of this git repo ... And we can fix all usages in this repo.